c/httpserver.c

```diff
@@ -3761,30 +3761,43 @@ void respondWithUnixFile(HttpResponse* response, char* absolutePath, int jsonMod
 
 // Response must ALWAYS be finished on return
 void respondWithUnixDirectory(HttpResponse *response, char* absolutePath, int jsonMode) {
+  int returnCode;
+  int reasonCode;
+    UnixFile *directory = NULL;
+
+  
 #ifdef DEBUG
   printf("Directory case: %s\n",absolutePath);
   fflush(stdout);
 #endif
-  setResponseStatus(response,200,"OK");
-  addStringHeader(response, "Cache-control", "no-store");
-  addStringHeader(response, "Pragma", "no-cache");  
-  addStringHeader(response,"Server","jdmfws");
-  addStringHeader(response,"Transfer-Encoding","chunked");
-  if (jsonMode == 0) {
-    setContentType(response,"text/html");
-    writeHeader(response);
-    StringListElt *parsedFileTail = firstStringListElt(response->request->parsedFile);
-    while (parsedFileTail->next){
-      parsedFileTail = parsedFileTail->next;
-    }
-    makeHTMLForDirectory(response, absolutePath, parsedFileTail->string,TRUE);
+  
+ if (directory = (directoryOpen(absolutePath,&returnCode,&reasonCode)) == NULL){
```

Comment:
* This produces a warning message
```
[exec] WARNING CCN3068 /u/tsirf/zowe-secmgmt/zowe-common-c/c/httpserver.c:3774  Operation between types "struct UnixFileStream_tag*" and "int" is not allowed.
```
* The if condition seems to have a bug - first we compare the result of ```dorectoryOpen``` with NULL and then assign the result (either 1 or 0 depending on the result) to ```directory```. I think you meant the following:
```
if ((directory = directoryOpen(absolutePath,&returnCode,&reasonCode)) == NULL){
```